patches/@mui+monorepo+5.10.16.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/node_modules/@mui/monorepo/packages/typescript-to-proptypes/src/generator.ts b/node_modules/@mui/monorepo/packages/typescript-to-proptypes/src/generator.ts
+index 2e243512..6314afe4 100644
+--- a/node_modules/@mui/monorepo/packages/typescript-to-proptypes/src/generator.ts
++++ b/node_modules/@mui/monorepo/packages/typescript-to-proptypes/src/generator.ts
+@@ -224,6 +224,17 @@ export function generate(component: t.Component, options: GenerateOptions = {}):
+         (type): type is t.LiteralType => type.type === 'LiteralNode',
+       );
+ 
++      if (context.propTypeDefinition.name === 'type') {
```

Comment:
I'm open to suggestions on how to tackle this one, because it also might be not stable (could no longer work if we update `mui/monorepo`).
Introducing this or similar change in base repo COULD be an option, but on the other hand, they technically don't care about such change. 🙈 🤷

Reply:
It may not be the most ideal fix as it will not allow any unknown string as `type`, which is the reason for `(string & {})` exists... 🤔 
This change avoid the following type madness due to the `(string & {})` hack:
**Before:**
```js
type: PropTypes.oneOfType([
  PropTypes.oneOf([
    'button',
    'checkbox',
    'color',
    'date',
    'datetime-local',
    'email',
    'file',
    'hidden',
    'image',
    'month',
    'number',
    'password',
    'radio',
    'range',
    'reset',
    'search',
    'submit',
    'tel',
    'text',
    'time',
    'url',
    'week',
  ]),
  PropTypes.shape({
    '__@iterator@17874': PropTypes.func.isRequired,
    anchor: PropTypes.func.isRequired,
    at: PropTypes.func.isRequired,
    big: PropTypes.func.isRequired,
    blink: PropTypes.func.isRequired,
    bold: PropTypes.func.isRequired,
    charAt: PropTypes.func.isRequired,
    charCodeAt: PropTypes.func.isRequired,
    codePointAt: PropTypes.func.isRequired,
    concat: PropTypes.func.isRequired,
    endsWith: PropTypes.func.isRequired,
    fixed: PropTypes.func.isRequired,
    fontcolor: PropTypes.func.isRequired,
    fontsize: PropTypes.func.isRequired,
    includes: PropTypes.func.isRequired,
    indexOf: PropTypes.func.isRequired,
    italics: PropTypes.func.isRequired,
    lastIndexOf: PropTypes.func.isRequired,
    length: PropTypes.number.isRequired,
    link: PropTypes.func.isRequired,
    localeCompare: PropTypes.func.isRequired,
    match: PropTypes.func.isRequired,
    matchAll: PropTypes.func.isRequired,
    normalize: PropTypes.func.isRequired,
    padEnd: PropTypes.func.isRequired,
    padStart: PropTypes.func.isRequired,
    repeat: PropTypes.func.isRequired,
    replace: PropTypes.func.isRequired,
    replaceAll: PropTypes.func.isRequired,
    search: PropTypes.func.isRequired,
    slice: PropTypes.func.isRequired,
    small: PropTypes.func.isRequired,
    split: PropTypes.func.isRequired,
    startsWith: PropTypes.func.isRequired,
    strike: PropTypes.func.isRequired,
    sub: PropTypes.func.isRequired,
    substr: PropTypes.func.isRequired,
    substring: PropTypes.func.isRequired,
    sup: PropTypes.func.isRequired,
    toLocaleLowerCase: PropTypes.func.isRequired,
    toLocaleUpperCase: PropTypes.func.isRequired,
    toLowerCase: PropTypes.func.isRequired,
    toString: PropTypes.func.isRequired,
    toUpperCase: PropTypes.func.isRequired,
    trim: PropTypes.func.isRequired,
    trimEnd: PropTypes.func.isRequired,
    trimLeft: PropTypes.func.isRequired,
    trimRight: PropTypes.func.isRequired,
    trimStart: PropTypes.func.isRequired,
    valueOf: PropTypes.func.isRequired,
  }),
]),
```

**After:**
```js
type: PropTypes.oneOf([
  'button',
  'checkbox',
  'color',
  'date',
  'datetime-local',
  'email',
  'file',
  'hidden',
  'image',
  'month',
  'number',
  'password',
  'radio',
  'range',
  'reset',
  'search',
  'submit',
  'tel',
  'text',
  'time',
  'url',
  'week',
]),
```

Reply:
I would propose: https://github.com/mui/mui-x/pull/7225#discussion_r1054889927.

Reply:
That is a good fallback to be aware of if we'll need it again. Thank you. 🙏